test/dsl/dsl_model_construction.jl

```diff
@@ -122,17 +123,16 @@ let
     push!(identical_networks_1, reaction_networks_standard[8] => different_arrow_8)
 
     for networks in identical_networks_1
-        f1 = ODEFunction(convert(ODESystem, networks[1]), jac = true)
-        f2 = ODEFunction(convert(ODESystem, networks[2]), jac = true)
-        g1 = SDEFunction(convert(SDESystem, networks[1]))
-        g2 = SDEFunction(convert(SDESystem, networks[2]))
         for factor in [1e-2, 1e-1, 1e0, 1e1, 1e2, 1e3]
-            u0 = factor * rand(rng, length(get_unknowns(networks[1])))
-            p = factor * rand(rng, length(get_ps(networks[1])))
+            u0_1 = rnd_u0(networks[1], rng; factor)
+            p_1 = rnd_ps(networks[1], rng; factor)
+            u0_2 = Pair.(unknowns(networks[2]), last.(u0_1))
+            p_2 = Pair.(parameters(networks[2]), last.(p_1))
             t = rand(rng)
-            @test all(abs.(f1(u0, p, t) .≈ f2(u0, p, t)))
-            @test all(abs.(f1.jac(u0, p, t) .≈ f2.jac(u0, p, t)))
-            @test all(abs.(g1(u0, p, t) .≈ g2(u0, p, t)))
+            
+            @test f_eval(networks[1], u0_1, p_1, t) ≈ f_eval(networks[2], u0_2, p_2, t)
```

Comment:
Uses the new `f_eval` function to evaluate the ODESystem `f` function.

Reply:
Alsmot always, like here, `f_eval` is only called once for each network, do no use using a in-place version. Because most networks have different `du` vector sizes, reusing it across different systems is hard.